### PR TITLE
[Feature] Add Rerange transform

### PR DESCRIPTION
**Pull request by yamengxi**

### CODE REVIEW

File: mmseg/datasets/pipelines/transforms.py [hunk 1]
```diff
@@ -390,6 +390,54 @@ def __repr__(self):
...
+        img_min_value = np.min(img)
+        img_max_value = np.max(img)
+        # rerange to [0, 1]
+        img = (img - img_min_value) / (img_max_value - img_min_value)
+        # rerange to [min_value, max_value]
+        img = img * (self.max_value - self.min_value) + self.min_value
+        results['img'] = img
+
+        results['img_rerange_cfg'] = dict(
+            min_value=self.min_value, max_value=self.max_value)
```
Comment (xvjiarui) [inline mmseg/datasets/pipelines/transforms.py:432]: What's the purpose of `img_rerange_cfg`?

**Comment by hellock** [comment]
Task linked: [CU-fjrgpn Data preprocessing](https://app.clickup.com/t/fjrgpn)

File: mmseg/datasets/pipelines/transforms.py [hunk 2]
```diff
@@ -390,6 +390,52 @@ def __repr__(self):
...
+        self.max_value = max_value
+
+    def __call__(self, results):
+        """Call function to rerange images.
+
+        Args:
+            results (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: Reranged results, 'img_rerange_cfg' key is added into
```
Comment (xvjiarui) [inline mmseg/datasets/pipelines/transforms.py:418]: Not updated

Comment (yamengxi) [reply mmseg/datasets/pipelines/transforms.py:418]: Sorry, I forget it!

File: mmseg/datasets/pipelines/transforms.py [hunk 3]
```diff
@@ -390,6 +390,52 @@ def __repr__(self):
...
+                result dict.
+        """
+
+        img = results['img']
+        img_min_value = np.min(img)
+        img_max_value = np.max(img)
+        # rerange to [0, 1]
+        img = (img - img_min_value) / (img_max_value - img_min_value)
```
Comment (xvjiarui) [inline mmseg/datasets/pipelines/transforms.py:426]: If we are reading an image with range [0.1, 1] (no pixel is black), 0.1 will be mapped to `min_value` (default 0), which is not ideal.

Comment (yamengxi) [reply mmseg/datasets/pipelines/transforms.py:425]: That's exactly what it's going to do.

**Review by xvjiarui — APPROVED** [review]

**Pull request merged**